src/model/element.js

```diff
@@ -116,13 +116,14 @@ export default class Element extends Node {
 	 * @returns {Boolean}
 	 */
 	is( type, name = null ) {
-		const cutType = type.replace( /^model:/, '' );
-
 		if ( !name ) {
-			return cutType == 'element' || cutType == this.name || super.is( type );
-		} else {
-			return cutType == 'element' && name == this.name;
+			return type === 'element' || type === 'model:element' ||
```

Comment:
One more optimization that you could consider is ordering those conditions in the likelihood of them being executed. You may need to do some stats and it's a subject of change, so we may need to repeat this. But if it yields nice results, it may be worth doing.

Reply:
OK, I'll check this.  
  
Side note I've done some quick (I don't have the results saved though) check for the fastest 'false' path. I've counted the checks in the `RootElement` checks and one of the most used was `is( 'text' )` . **But** adding `if( type === 'text' ) return false` (fail fast) turned out to yield be slower than proposed solution in PR. It might be some browser optimization or a simple error in my methodology but this proves that intuition must be followed with tests.

Reply:
I don't think that this will yield any significant result. I've done some tests and below are number of executions of some `is` checks in the long semanting + undo scenario:

![](https://user-images.githubusercontent.com/247363/78558992-e16dc280-7813-11ea-89ef-9fc9630eb250.png)

As you can see there are milions of checks using `is()` methods and most of them checks if element is `text` or `element`. But if you check how much time is spent (self-time) in `is` methods then in the above scenario (~20 seconds of undo execution) they add up to maybe 150 - 200ms:

![](https://user-images.githubusercontent.com/247363/78559219-49bca400-7814-11ea-8f7b-d950db0d602c.png)

I don't see how we could optimize those paths reliably. At this stage I would try to